addons/calendar/models/calendar_recurrence.py

```diff
@@ -309,6 +309,10 @@ def _write_events(self, values, dtstart=None):
         events = self._get_events_from(dtstart) if dtstart else self.calendar_event_ids
         return events.with_context(no_mail_to_attendees=True, dont_notify=True).write(dict(values, recurrence_update='self_only'))
 
+    def _unlink_events(self, dtstart=None):
+        events = self._get_events_from(dtstart) if dtstart else self.calendar_event_ids
+        return events.unlink()
+
```

Comment:
There is already a `action_mass_deletion` method on calendar_event. It takes the event as an argument and it can delete after the event start date of the current one if needed. I think the code should be simplfied to either have only one method or call _unlink_events in action_mass_deletion.